implementations/rust/ockam/ockam_core/src/routing/address.rs

```diff
@@ -28,11 +31,16 @@ use serde::{Deserialize, Serialize};
 /// * `"0#alice"` represents a local worker with the address: `alice`.
 /// * `"1#carol"` represents a remote worker with the address `carol`, reachable over TCP transport.
 ///
-#[derive(Serialize, Deserialize, Clone, Hash, Ord, PartialOrd, Eq, PartialEq)]
+#[derive(Serialize, Deserialize, Decode, Encode, Clone, Hash, Ord, PartialOrd, Eq, PartialEq)]
+#[rustfmt::skip]
+#[cbor(map)]
 pub struct Address {
```

Comment:
In order to use `Route` and **not** `MultiAddr` I added the Route ad as a cbor encodable.
We can express a `Route` as `MultiAddr` but we need to double cast it, i think sometimes a `Route`  is really what we want.